### PR TITLE
Change scio-smb to depend on provided scio io modules

### PR DESCRIPTION
**Pull request by RustedBones**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1316,8 +1315,11 @@ lazy val `scio-smb`: Project = project
   .in(file("scio-smb"))
   .dependsOn(
     `scio-core`,
-    `scio-test` % "test;it",
-    `scio-avro` % IntegrationTest
+    `scio-avro` % Provided,
+    `scio-google-cloud-platform` % Provided,
```
Comment (RustedBones) [inline build.sbt:1319]: smb json is using BQ `TableRow`

File: build.sbt [hunk 2]
```diff
@@ -1340,29 +1341,27 @@ lazy val `scio-smb`: Project = project
...
-      "org.apache.avro" % "avro" % avroVersion % Provided,
-      "org.apache.hadoop" % "hadoop-common" % hadoopVersion % Provided,
-      "org.apache.parquet" % "parquet-avro" % parquetVersion % Provided excludeAll ("org.apache.avro" % "avro"),
-      "org.apache.parquet" % "parquet-column" % parquetVersion % Provided,
-      "org.apache.parquet" % "parquet-common" % parquetVersion % Provided,
-      "org.apache.parquet" % "parquet-hadoop" % parquetVersion % Provided,
-      "org.tensorflow" % "tensorflow-core-api" % tensorFlowVersion % Provided,
-      // runtime
-      "org.apache.beam" % "beam-sdks-java-io-hadoop-format" % beamVersion % Runtime,
-      "org.apache.hadoop" % "hadoop-client" % hadoopVersion % Runtime,
```
Comment (RustedBones) [inline build.sbt:1365]: No need to define runtime deps anymore, since the'll be pulled by the provided modules

File: build.sbt [hunk 3]
```diff
@@ -1340,29 +1341,27 @@ lazy val `scio-smb`: Project = project
...
+      "org.apache.avro" % "avro" % avroVersion % Provided, // scio-avro
+      "org.apache.beam" % "beam-sdks-java-extensions-avro" % beamVersion % Provided, // scio-avro
+      "org.apache.beam" % "beam-sdks-java-extensions-protobuf" % beamVersion % Provided, // scio-tensorflow
+      "org.apache.beam" % "beam-sdks-java-io-google-cloud-platform" % beamVersion % Provided, // scio-gcp
+      "org.apache.beam" % "beam-sdks-java-io-hadoop-common" % beamVersion % Provided, // scio-parquet
+      "org.apache.hadoop" % "hadoop-common" % hadoopVersion % Provided, // scio-parquet
+      "org.apache.parquet" % "parquet-avro" % parquetVersion % Provided excludeAll ("org.apache.avro" % "avro"), // scio-parquet
+      "org.apache.parquet" % "parquet-column" % parquetVersion % Provided, // scio-parquet
+      "org.apache.parquet" % "parquet-common" % parquetVersion % Provided, // scio-parquet
+      "org.apache.parquet" % "parquet-hadoop" % parquetVersion % Provided, // scio-parquet
```
Comment (RustedBones) [inline build.sbt:1363]: added comment for required compile provided dep, indicating the scio module origin

**Review by clairemcginty — COMMENTED** [review]
Looks good, maybe we should update https://spotify.github.io/scio/extras/Sort-Merge-Bucket.html to explain the change/how users should import these libs

**Review by clairemcginty — APPROVED** [review]

File: build.sbt (hunk 1 above)
Comment (kellen) [reply build.sbt:1319]: Wut

File: build.sbt [hunk 4]
```diff
@@ -1046,6 +1046,9 @@ lazy val `scio-parquet` = project
       "org.slf4j" % "slf4j-api" % slf4jVersion,
       // provided
       "org.tensorflow" % "tensorflow-core-api" % tensorFlowVersion % Provided,
+      // runtime
+      "org.apache.hadoop" % "hadoop-client" % hadoopVersion % Runtime excludeAll (Exclude.metricsCore),
```
Comment (RustedBones) [inline build.sbt:1050]: This dep is declared as provided in beam parquet io [here](https://github.com/apache/beam/blob/e6e180b68965e872822f5f32281b30c65da16af3/sdks/java/io/parquet/build.gradle#L51C3-L51C11).

Moved the dependency from `scio-smb` to `scio-parquet`

**Pull request merged**